### ISSUE TITLE
libuv flooded callback on server disconnect

### ISSUE THREAD

**Issue by joagr** [CopernicaMarketingSoftware/AMQP-CPP#155]
Using libuv on linux, I'm finding that a disconnect from the server results in 100% CPU usage - the callback is flooded with "readable" events. Is there a good way to catch this?

**Comment by EmielBruijntjes** [comment]
Do you have an example program? I made a simple example, but that did not cause any issues when I closed the connection via the RabbitMQ web interface.

**Comment by joagr** [comment]
Hi Emiel, thank you for the sample program. I used it for testing in two different environments, and found that the issue only happens on my devel environment, which is Windows Subsystem for Linux (WSL / bash on Windows). When I ran the test on a straight linux machine, I did not see the issue.

For my purposes, we can close this issue#. I only use WSL for development.

A couple of notes about reproducing the issue on WSL, in case they are helpful to anyone:

1. I found that closing the connection via the web interface did not raise the issue. I was simulating a server crash or network drop by stopping the Docker container that my Rabbit test server runs in.

2. The sample program, as is, halts when the server dies. To reproduce the issue, I had to keep the event loop going. I did that with a simple timer:

// Define a timer to keep the event loop ticking even if the rabbit connection drops
void TimerCallback(uv_timer_t * timerHandle) {
               // Restart the timer
               uv_timer_start(timerHandle, TimerCallback, 2000, 0);
}

and in main(), before uv_run:

               // Define a timer to keep the event loop ticking even if the rabbit connection drops
               uv_timer_t uvTimer;
               uv_timer_init(loop, &uvTimer);
               uv_timer_start(&uvTimer, TimerCallback, 2000, 0);
